### PR TITLE
Issue #176:  Migration of AmenityController to OpenApi yaml schema

### LINKED ISSUE #176: Migration of AmenityController to OpenApi yaml schema
This task is a subtask for #169.

The steps that need to be done:
- convert all endpoints from controller to yaml file
- check that endpoints are working, in the same way, like before the change
- check the tests

An example can be found in PR #171.

### PR DESCRIPTION
**Pull request by SergeyHydra**
🚀 Description

Added OpenAPI spec for AmenityController

📄 Motivation and Context

Fixes Issue #176 

🧪 How Has This Been Tested?

Test cases are passing
Endpoints are working
📷 Screenshots (if appropriate)

📦 Types of changes

  Bug fix (non-breaking change which fixes an issue)
  New feature (non-breaking change which adds functionality)
  Breaking change (fix or feature that would cause existing functionality to not work as expected)
✅ Checklist

  My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
  My change requires a change to the documentation.
  I have updated the documentation accordingly.

### CODE REVIEW

File: service/src/main/java/com/myhome/controllers/AmenityController.java [hunk 1]
```diff
@@ -16,46 +16,37 @@
...
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
+@Slf4j
 @RequiredArgsConstructor
-public class AmenityController {
+public class AmenityController implements AmenitiesApi, CommunitiesApi {
```
Comment (mslowiak) [inline service/src/main/java/com/myhome/controllers/AmenityController.java]: `AmenityController ` should only implement `AmenitiesApi`

File: api/src/main/resources/api.yaml [hunk 2]
```diff
@@ -10,6 +10,135 @@ tags:
   - name: Documents
   - name: Communities
 paths:
+  /amenities/{amenityId}:
+    get:
+      tags:
+        - Amentities
```
Comment (mslowiak) [inline api/src/main/resources/api.yaml:20]: `Amentities` tag is not added in line 11 of this file

File: api/src/main/resources/api.yaml [hunk 3]
```diff
@@ -437,6 +566,54 @@ paths:
...
+          type: string
+        price:
+          type: long
+        communityId:
+          type: string
+    AmenityDto:
+      type: object
+      properties:
+        id:
+          type: long
```
Comment (mslowiak) [inline api/src/main/resources/api.yaml]: long is incorrect in OpenApi specification, please use:
```
type: integer
format: int64
```

**Review by mslowiak — CHANGES_REQUESTED** [review]
The build is failing because of non implemented methods in CommunityController

**Comment by mslowiak** [comment]
Please use rebase over merge and fix the pull request name

**Comment by SergeyHydra** [comment]
@mslowiak can I move methods with community path from AmenityController to CommunityController?

**Review by mslowiak — CHANGES_REQUESTED** [review]
Looks good for me but please use rebase over merge. 
Ping me when it will be fixed so we can merge it

**Comment by mslowiak** [comment]
@SergeyHydra 
Looks like the build is failing

**Comment by jmprathab** [comment]
@SergeyHydra Please fix the build and resolve the conflicts. Once done this is ready to merge.

**Review by mslowiak — APPROVED** [review]

**Pull request merged**